### ISSUE TITLE
No intercepts (bias) for Poisson/Tweedie/etc.

### ISSUE THREAD

**Issue by david-cortes** [catboost/catboost#2852]
Trying to fit regression models which involve link functions, such as Poisson or Tweedie, results in CatBoost not estimating any intercept for them, nor allowing to pass `boost_from_average`:

This can lead to results that are very off from the actual values. Example:
```python
import numpy as np
from catboost import CatBoostRegressor
from sklearn.datasets import make_regression
X, y = make_regression(n_samples=10, n_features=2, random_state=123)
y = np.exp((y - y.mean()) / y.std() + 0.3)
model = CatBoostRegressor(
    objective="Tweedie:variance_power=1.99",
    iterations=2
).fit(X, y)
model.predict(X)
```
```
array([1.04579494, 1.0186729 , 0.99907772, 1.0026361 , 1.01568892,
       0.98600012, 0.99556374, 1.00290618, 1.0115019 , 1.0115019 ])
```
```python
y
```
```
array([12.56959998,  2.58808778,  1.03713092,  0.74657393,  1.9183389 ,
        0.23660995,  0.76927224,  0.67357035,  2.07876872,  1.63098162])
```

For the most part, it should be possible to easily and quickly estimate the optimal intercepts for most of the regression objects in CatBoost - see for example:
https://github.com/dmlc/xgboost/pull/10298

Also BTW: MAE should be initialized from the median rather than the average.

**Comment by andrey-khropov** [comment]
> Also BTW: MAE should be initialized from the median rather than the average.

Yes. And it is in fact initialized from the median as you can see from [the code](https://github.com/catboost/catboost/blob/fff94d7dab80ed69aea2e780f6a9554ef6d741bb/catboost/libs/metrics/optimal_const_for_loss.h#L204).

The parameter name `boost_from_average` was taken from `LightGBM` where this functionality has been implemented first to aid an interoperability between different GBDT packages. 

But its meaning has expanded since the generalization to other loss functions.

The documentation correctly states that it means "the best constant value for the specified loss function".

Perhaps we should introduce a synonym for this parameter (like `boost_from_optimal_constant`) that better reflects this to avoid confusion.

**Comment by david-cortes** [comment]
> > Also BTW: MAE should be initialized from the median rather than the average.
> 
> Yes. And it is in fact initialized from the median as you can see from [the code](https://github.com/catboost/catboost/blob/fff94d7dab80ed69aea2e780f6a9554ef6d741bb/catboost/libs/metrics/optimal_const_for_loss.h#L204).
> 
> The parameter name `boost_from_average` was taken from `LightGBM` where this functionality has been implemented first to aid an interoperability between different GBDT packages.
> 
> But its meaning has expanded since the generalization to other loss functions.
> 
> The documentation correctly states that it means "the best constant value for the specified loss function".
> 
> Perhaps we should introduce a synonym for this parameter (like `boost_from_optimal_constant`) that better reflects this to avoid confusion.

But the 'bias' is set to zero when using an objective with a link function like Poisson or Tweedie.

**Comment by andrey-khropov** [comment]
> But the 'bias' is set to zero when using an objective with a link function like Poisson or Tweedie.

Yes. I'm not closing this issue.
